source/slang/slang-compiler.cpp

```diff
@@ -418,18 +418,26 @@ namespace Slang
             }
             case CodeGenTarget::GLSL:
             {
-                // Can always output GLSL
-                return PassThroughMode::None; 
+                // For the prelude we'll use Glslang
+                return PassThroughMode::Glslang;
             }
             case CodeGenTarget::HLSL:
             {
-                // Can always output HLSL
-                return PassThroughMode::None;
+                // Use the default compiler for the source language if set, 
+                DownstreamCompiler* downstreamCompiler = session->getDefaultDownstreamCompiler(SourceLanguage::HLSL);
+                if (downstreamCompiler)
+                {
+                    return PassThroughMode(downstreamCompiler->getDesc().type);
+                }
+                else
+                {
+                    // This is ambiguous, because we could use dxc or fxc. For now we'll go with Dxc. 
+                    return PassThroughMode::Dxc;
```

Comment:
Do we eventually need to add `PassThroughMode::GenericHLSL` and then have it select between fxc/dxc based on the shader model?